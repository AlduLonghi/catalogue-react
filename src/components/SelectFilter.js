import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import changeFilter from '../redux/actions/selectFilter';

const SelectFilter = ({ changeFilter }) => {
  const values = ['All', 'Painting', 'Metalwork', 'Sculpture', 'Ceramic', 'Textile', 'Miscellaneous'];

  const handleChange = e => {
    changeFilter(e.target.value);
  };

  return (
    <select data-testid="select" className="form-select" onChange={handleChange}>
      {values.map(val => <option key={val} value={val}>{val}</option>)}
    </select>
  );
};

SelectFilter.propTypes = {
  changeFilter: PropTypes.func.isRequired,
};

const mapDispatchToProps = {
  changeFilter,
};

export default connect(null, mapDispatchToProps)(SelectFilter);
