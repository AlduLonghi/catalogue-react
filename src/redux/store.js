import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from './reducers/index';

const initialState = {};

const store = createStore(rootReducer, initialState, applyMiddleware(thunk));

export default store;
