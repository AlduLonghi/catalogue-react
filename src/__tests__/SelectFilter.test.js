import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { render, fireEvent, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import SelectFilter from '../components/SelectFilter';
import store from '../redux/store';

it('Renders SelectFilter Component', () => {
  const filter = renderer.create(
    <Provider store={store}>
      <SelectFilter />
    </Provider>,
  ).toJSON();
  expect(filter).toMatchSnapshot();
});

it('Should change the value when you click an option', () => {
  render(
    <Provider store={store}>
      <SelectFilter />
    </Provider>,
  );
  expect(screen.getByText('All')).toBeInTheDocument();

  fireEvent.change(screen.getByTestId('select'), {
    target: { value: 'Painting' },
  });

  expect(screen.getByText('Painting')).toBeInTheDocument();
});
