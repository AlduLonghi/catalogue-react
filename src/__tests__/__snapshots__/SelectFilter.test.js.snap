// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders SelectFilter Component 1`] = `
<select
  className="form-select"
  data-testid="select"
  onChange={[Function]}
>
  <option
    value="All"
  >
    All
  </option>
  <option
    value="Painting"
  >
    Painting
  </option>
  <option
    value="Metalwork"
  >
    Metalwork
  </option>
  <option
    value="Sculpture"
  >
    Sculpture
  </option>
  <option
    value="Ceramic"
  >
    Ceramic
  </option>
  <option
    value="Textile"
  >
    Textile
  </option>
  <option
    value="Miscellaneous"
  >
    Miscellaneous
  </option>
</select>
`;
